SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -938,31 +962,29 @@ void USpatialReceiver::HandleUnreliableRPC(Worker_ComponentUpdateOp& Op)
 	{
 		Schema_Object* EventData = Schema_IndexObject(EventsObject, SpatialConstants::UNREAL_RPC_ENDPOINT_EVENT_ID, i);
 
-		uint32 Offset = Schema_GetUint32(EventData, SpatialConstants::UNREAL_RPC_PAYLOAD_OFFSET_ID);
-		uint32 Index = Schema_GetUint32(EventData, SpatialConstants::UNREAL_RPC_PAYLOAD_RPC_INDEX_ID);
-		TArray<uint8> PayloadData = GetBytesFromSchema(EventData, SpatialConstants::UNREAL_RPC_PAYLOAD_RPC_PAYLOAD_ID);
-		int64 CountBits = PayloadData.Num() * 8;
+		RPCPayload Payload(EventData);
 
-		FUnrealObjectRef ObjectRef(EntityId, Offset);
+		FUnrealObjectRef ObjectRef(EntityId, Payload.Offset);
 
 		UObject* TargetObject = PackageMap->GetObjectFromUnrealObjectRef(ObjectRef).Get();
 
 		if (!TargetObject)
 		{
-			UE_LOG(LogSpatialReceiver, Warning, TEXT("HandleUnreliableRPC: Could not find target object: %s, skipping rpc at index: %d"), *ObjectRef.ToString(), Index);
+			UE_LOG(LogSpatialReceiver, Warning, TEXT("HandleUnreliableRPC: Could not find target object: %s, skipping rpc at index: %d"), *ObjectRef.ToString(), Payload.Index);
 			continue;
 		}
 
 		const FClassInfo& ClassInfo = ClassInfoManager->GetOrCreateClassInfoByObject(TargetObject);
 
-		UFunction* Function = ClassInfo.RPCs[Index];
-		ApplyRPC(TargetObject, Function, PayloadData, CountBits, FString());
+		UFunction* Function = ClassInfo.RPCs[Payload.Index];
+		ApplyRPC(TargetObject, Function, Payload, FString());
 	}
 }
 
 void USpatialReceiver::OnCommandRequest(Worker_CommandRequestOp& Op)
 {
 	Schema_FieldId CommandIndex = Schema_GetCommandRequestCommandIndex(Op.request.schema_type);
+	ensure(CommandIndex == SpatialConstants::UNREAL_RPC_ENDPOINT_COMMAND_ID);
```

Comment:
Move this ensure below the ifs for commands on specific components.

Reply:
Removed ensure. Added PLAYER_SPAWNER_COMMAND_ID that is used in comparison now for readability.